app/views/papers/_index.html.erb

```diff
@@ -52,8 +52,13 @@
         <% end %>
       </small>
       <ul class="nav nav-pills" role="tablist">
+        <% comments_count = paper_reference ? paper_reference.comments.map {|c| c.self_and_descendants.length}.inject(0, :+).to_s : paper.references.map { |r| r.comments.map {|c| c.self_and_descendants.length}.inject(0, :+)}.inject(0, :+).to_s %>
         <% %i{abstract details comments}.each do |tab| %>
-          <li><%= link_to tab.capitalize, '', "data-detail-id" => "#{tab}-#{index}", class: "small paper-detail-tab" %></li>
+          <% if tab == :comments %>
+            <li><%= link_to tab.capitalize.to_s + "(" + comments_count + ")", '', "data-detail-id" => "#{tab}-#{index}", class: "small paper-detail-tab" %></li>
+          <% else %>
+            <li><%= link_to tab.capitalize, '', "data-detail-id" => "#{tab}-#{index}", class: "small paper-detail-tab" %></li>
+          <% end %>
```

Comment:
Avoid writing additional code where possible. It's also kind of redundant in the most complicated part of the line. Instead of:

```erb
<% if tab == :comments %>
  <li><%= link_to tab.capitalize.to_s + "(" + comments_count + ")", '', "data-detail-id" => "#{tab}-#{index}", class: "small paper-detail-tab" %></li>
<% else %>
  <li><%= link_to tab.capitalize, '', "data-detail-id" => "#{tab}-#{index}", class: "small paper-detail-tab" %></li>
<% end %>
```
You could do something like:
```erb
<% 
  tab_link_text = tab.capitalize
  tab_link_text += " (#{comment_count})" if tab == :comment
%>
<li><%= link_to tab_link_text, '', "data-detail-id" => "#{tab}-#{index}", class: "small paper-detail-tab" %></li>
```